ddl/db_integration_test.go

```diff
@@ -743,6 +743,31 @@ func (s *testIntegrationSuite4) TestChangingTableCharset(c *C) {
 			") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin",
 	))
 
+	tk.MustExec("drop table t")
+	tk.MustExec("create table t(a varchar(5) charset utf8) charset utf8")
+	tk.MustExec("alter table t charset utf8mb4")
+	tbl = testGetTableByName(c, s.ctx, "test", "t")
+	c.Assert(tbl, NotNil)
+	c.Assert(tbl.Meta().Charset, Equals, "utf8mb4")
+	c.Assert(tbl.Meta().Collate, Equals, "utf8mb4_bin")
+	for _, col := range tbl.Meta().Columns {
+		// Column charset and collate should remain unchanged.
+		c.Assert(col.Charset, Equals, "utf8")
+		c.Assert(col.Collate, Equals, "utf8_bin")
+	}
+
+	tk.MustExec("drop table t")
+	tk.MustExec("create table t(a varchar(5) charset utf8 collate utf8_unicode_ci) charset utf8 collate utf8_unicode_ci")
+	tk.MustExec("alter table t collate utf8_danish_ci")
+	tbl = testGetTableByName(c, s.ctx, "test", "t")
+	c.Assert(tbl, NotNil)
+	c.Assert(tbl.Meta().Charset, Equals, "utf8")
+	c.Assert(tbl.Meta().Collate, Equals, "utf8_danish_ci")
+	for _, col := range tbl.Meta().Columns {
+		c.Assert(col.Charset, Equals, "utf8")
+		// Column collate should remain unchanged.
+		c.Assert(col.Collate, Equals, "utf8_unicode_ci")
```

Comment:
Maybe you should add a case for `convert to`—— both table and column's charset are changed.

Reply:
The case already exists:
https://github.com/pingcap/tidb/blob/d5be4b6c485ec80cb5acd69dcca71feefd33433b/ddl/db_integration_test.go#L657-L671

Reply:
got